docs/troubleshooting.md

```diff
@@ -0,0 +1,199 @@
+    kubernetes-incubator/service-catalog
+  Generation:           1
+  Resource Version:     21
+  Self Link:            /apis/servicecatalog.k8s.io/v1alpha1/servicebrokers/ansible-service-broker
+  UID:                  8f3782b5-a9f1-11e7-ab6a-0242ac110006
+Spec:
+  Auth Info:
+    Bearer:
+      Secret Ref:
+        Kind:           Secret
+        Name:           ansibleservicebroker-client
+        Namespace:      ansible-service-broker
+  Relist Behavior:      Duration
+  Relist Duration:      15m0s
+  Relist Requests:      0
+  URL:                  https://asb.ansible-service-broker.svc:1338/ansible-service-broker/
+Status:
+  Conditions:
+    Last Transition Time:       2017-10-05T17:22:01Z
+    Message:                    Error fetching catalog. Error getting broker catalog for broker "ansible-service-broker": Get https://asb.ansible-service-broker.svc:1338/ansible-service-broker/v2/catalog: x509: certificate signed by unknown authority
+    Reason:                     ErrorFetchingCatalog
+    Status:                     False
+    Type:                       Ready
+  Operation Start Time:         2017-10-05T17:22:02Z
+  Reconciled Generation:        0
+Events:
+  FirstSeen     LastSeen        Count   From                                    SubObjectPath   Type            Reason                  Message
+  ---------     --------        -----   ----                                    -------------   --------        ------                  -------
+  6s            6s              1       service-catalog-controller-manager                      Warning         ErrorFetchingCatalog    Error getting broker catalog for broker "ansible-service-broker": Get https://asb.ansible-service-broker.svc:1338/ansible-service-broker/v2/catalog: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
+  6s            2s              9       service-catalog-controller-manager                      Warning         ErrorFetchingCatalog    Error getting broker catalog for broker "ansible-service-broker": Get https://asb.ansible-service-broker.svc:1338/ansible-service-broker/v2/catalog: x509: certificate signed by unknown authority
+```
+
+#### Resolution: Provide caBundle to service-catalog
+
+We can get the caBundle with the following command:
+
+```
+$ oc get secret -n kube-service-catalog -o go-template='{{ range .items }}{{ if eq .type "kubernetes.io/service-account-token" }}{{ index .data "service-ca.crt" }}{{end}}{{"\n"}}{{end}}' | tail -n 1
+LS0t...really long string...
+```
+
+And update the servicebroker object, adding `caBundle` to the ansible-service-broker's `Spec`:
+
+```
+$ oc edit servicebroker ansible-service-broker
+servicebroker "ansible-service-broker" edited
+```
+
+### Problem: Service Catalog Invalid Credentials
+
+In the error condition message below you can see that the Service Catalog fails
+to get the catalog for the "ansible-service-broker" because of "invalid
+credentials":
```

Comment:
change:

If the service-catalog does not have the proper credentials, it will not be able to communicate with the broker. In the error message below, you see that the service-catalog fails because of `invalid credentials` error.

Reply:
I updated this to parallel the approach of the previous "problem".